File: mean/src/main/java/com/blog/mean/service/CommentService.java
package com.blog.mean.service;

import com.blog.mean.dto.CommentDTO;
import com.blog.mean.dto.CreateCommentRequest;
import java.util.List;

public interface CommentService {
    CommentDTO createComment(CreateCommentRequest request);
    List<CommentDTO> getCommentsByPost(Long postId);
    CommentDTO updateComment(Long id, String content);
    void deleteComment(Long id);
}

File: frontend/src/services/commentService.js
import api from './api';

export const commentService = {
  createComment: (commentData) => api.post('/comments', commentData),
  getCommentsByPost: (postId) => api.get(`/comments/post/${postId}`),
  updateComment: (id, content) => api.put(`/comments/${id}`, content),
  deleteComment: (id) => api.delete(`/comments/${id}`),
};